### ISSUE TITLE
Automate our GitHub related workflows

### ISSUE THREAD

**Issue by carmocca** [PyTorchLightning/pytorch-lightning#9140]
### Motivation

As the project and the team grows, we should start trying to standardize and automate how we work with GitHub to improve efficiency. We are reaching a limit in how we can check all our notifications.

### Pitch

Things to improve:
- Standardize the commit selection for bug-fix releases: This has been partly done in https://github.com/PyTorchLightning/pytorch-lightning/issues/7467 but could be further improved by automatically saving the partition of commits somewhere (an issue for it?).
- Standardize author list for releases: This has been partly done in https://github.com/PyTorchLightning/pytorch-lightning/issues/7467
- Subscribing to labels: We could follow what PyTorch does with https://github.com/pytorch/pytorch/issues/24422. A more specific list of labels would be necessary.
- Automatically assigning maintainers to pull requests depending on the labels or if they are CODEOWNERS.
- Automating triage review for the on-call schedule.

PyTorch deals with this by a custom `probot` implementation https://github.com/pytorch/pytorch-probot.

Alternatively, for a simpler but more manual solution, we could leverage the recently added `github CLI` extensions:
https://github.blog/2021-08-24-github-cli-2-0-includes-extensions/
Where we add extensions for each of these. The updates performed to issues/PRs by the extensions must be idempotent and would be manually called by us on demand. This avoids hosting concerns.

### Additional context

Please share any ideas, proposals, or questions on how we can make our life simpler.

cc @tchaton @ananthsub 

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>
